bzr.go

```diff
@@ -0,0 +1,78 @@
+package vcsstate
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/shurcooL/go/trim"
+)
+
+type bzr struct{}
+
+func (v bzr) DefaultBranch() string {
+	return v.defaultBranch()
+}
+
+func (bzr) Status(dir string) (string, error) {
+	cmd := exec.Command("bzr", "status")
+	cmd.Dir = dir
+
+	out, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	return string(out), nil
+}
+
+func (bzr) Branch(dir string) (string, error) {
+	cmd := exec.Command("bzr", "branch")
+	cmd.Dir = dir
+
+	out, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	return trim.LastNewline(string(out)), nil
+}
+
+// Bazaar uses UUID instead of SHA-1 hashes and is roughtly 60 characters in length
```

Comment:
I see.

In that case, doing `out[:bzrRevisionLength]` is probably not going to work and you'll need to implement that differently. Perhaps read until newline? Doing `out[:gitRevisionLength]` works for git since its hash is always the same length.